acceptance/suites/tests/00_smoke/puppetdb_integration.rb

```diff
@@ -10,7 +10,14 @@
 ## PuppetDB successfully received the agent's report sent from Puppet Server.
 ## We can just run the agent that's on the master for this.
 
-confine :to, {:platform => puppetdb_supported_platforms}, master
+# This confine was originally
+# confine :to, {:platform => puppetdb_supported_platforms}, master
+# With , master present, execution unexpectedly enters the body of the
+# test when we expect the test to be skipped.  BKR-650 maybe relavent.
+# Be aware that future modifications to this test might, especially
+# if there are test operations added that run against agents instead of just
+# masters.
+confine :to, {:platform => puppetdb_supported_platforms}
```

Comment:
@nwolfe Could you do a sanity check on this?  I'm thinking you put ", master" on there intentionally but I have no idea how any of this works.

 We found that this blew up with ...

```
<Beaker::DSL::Outcomes::FailTest: There should be one host with master defined
```

... if run against a test configuration matrix where no master from the test matrix matches a `puppetdb_supported_platforms` entry:

https://jenkins.puppetlabs.com/job/platform_puppetserver_integration-system_no-conditional_smoke-master/LAYOUT=ubuntu1510-64ma-64a,LDAP_TYPE=default,PLATFORM=default,label=beaker/18/consoleFull

Although ... it seems like this check has been sufficient to skip platforms previously until so, not sure.  Maybe this is even more broken with the newer beaker version that we bumped up to recently, 2.39.0.

Anyway, does this seem like the best way to work around the problem?